src/components/Files/LinkToSong/LinkToSong.container.js

```diff
@@ -17,23 +19,32 @@ const LinkToSongContainer = ({
 }) => {
   const songIndex = useSelector((state) => state.songsQueue.get('index'));
   const playingSongPath = useSelector((state) => selectSongPathAtIndex(state, songIndex));
-
+  const favorites = useSelector((state) => state.favorites.get('songs'));
   const dispatch = useDispatch();
 
+  const isFavorite = isSongInFavorites(favorites, path);
+  const isPlaying = path === playingSongPath;
+
   const selectSong = () => {
     dispatch(getSongStreamLink(path));
     dispatch(setSongIndex(index));
     dispatch(changeSongsQueue(files));
   };
 
-  const isPlaying = path === playingSongPath;
+  const handleFavorite = () => {
+    dispatch(handleInteractionWithFavorite(isFavorite, {
+      song_name: fileName,
+      path_lower: path,
```

Comment:
use camel case for variables, rename later as part of the marshal/unmarshal routine.